### PR TITLE
Add starter packs to resources page

### LINKED ISSUE #87: Add starter packs to new resources page
Create a new page for resources and add the starter packs to it according to the JSON file @rileysw provided

### PR DESCRIPTION
**Pull request by samderanova**
- Update navigation bar with new route to resources
- Add icons for each type of starter pack, following the design from graphics

### CODE REVIEW

File: src/assets/data/resources.json [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+				"tooltip": "Use Postman to test APIs!"
+			},
+			{
+				"name": "Ngrok",
+				"link": "https://ngrok.io",
+				"tooltip": "Instantly create a public URL for a website running locally on your development machine."
+			}
+		]
+	}
+]
```
[the remark this answers is not in the document]
Comment (samderanova) [reply src/assets/data/resources.json:184]: Uhh anyone have any idea what this means?

[the remark this answers is not in the document]
Comment (rileysw) [reply src/assets/data/resources.json:184]: The two files are the same on my end 😩

[the remark this answers is not in the document]
Comment (taesungh) [reply src/assets/data/resources.json:184]: Could be your line endings, i.e. CRLF vs LF

**Review by rileysw — DISMISSED** [review]

File: src/views/resources/Resource/Resource.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+				overlay={<Tooltip id={"resource-description"}>{tooltip}</Tooltip>}
+			>
+				<a
+					className={
+						(resourceIndex % 2 === 0
+							? styles["badge-gold"]
+							: styles["badge-dark"]) + " d-block my-3 p-3 rounded-pill badge"
+					}
+					href={link}
+					style={{ textDecoration: "none" }}
```
Comment (taesungh) [inline src/views/resources/Resource/Resource.tsx:29]: Thought [non-blocking]: using button styles might make this simpler, but I don't want to think about it right now.

File: src/views/resources/Resources.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import Head from "next/head";
+import resources from "assets/data/resources.json";
+import aiML from "assets/icons/ai-ml.svg";
+import backend from "assets/icons/backend.svg";
+import database from "assets/icons/database-fill.svg";
+import deploy from "assets/icons/deploy.svg";
+import userInterface from "assets/icons/display.svg";
+import gaming from "assets/icons/gaming.svg";
+import tools from "assets/icons/tools.svg";
```
Comment (taesungh) [inline src/views/resources/Resources.tsx:9]: Note: for future reference, might be nice to include the source of these icon assets as part of the commit message body.

**Review by taesungh — CHANGES_REQUESTED** [review]
Looks mostly good, small issue on the hover transition.

File: src/views/resources/Resource/Resource.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+interface ResourceProps {
+	resourceIndex: number;
+	name: string;
+	link: string;
+	tooltip: string;
+}
+
+function Resource({ resourceIndex, name, link, tooltip }: ResourceProps) {
+	const target = useRef(null);
```
[the remark this answers is not in the document]
Comment (RandyHuynh5815) [reply src/views/resources/Resource/Resource.tsx:14]: It's used in the a element within the OverlayTrigger.

File: src/views/resources/Resource/Resource.module.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
+@import "styles/gallery-theme";
+
+.badge-gold {
+	color: black;
+	background-color: $orange;
+
+	&:hover {
+		background-color: $light-orange;
+		transition: 0.25s ease;
```
[the remark this answers is not in the document]
Comment (taesungh) [reply src/views/resources/Resource/Resource.module.scss]: @RandyHuynh5815, we should also constrain the transition property

**Comment by lassenordahl** [comment]
This site is lovely! I have one request. Could you update the CockroachDB link to:

https://www.cockroachlabs.com/docs?utm_source=hackuci2023&utm_medium=event&utm_campaign=hackathon

so we could do some cross domain tracking for Hackathon users? That would be awesome.

File: src/views/resources/Resource/Resource.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (samderanova) [reply src/views/resources/Resource/Resource.tsx:14]: ^^

File: src/views/resources/Resources.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+const RESOURCE_INDEX_MAP: IndexMap = {
+	"User Interface": { svg: userInterface, color: "gold" } as IndexMapValue,
+	"Backend": { svg: backend, color: "blue" } as IndexMapValue,
+	"Databases": { svg: database, color: "gold" } as IndexMapValue,
+	"Deployment": { svg: deploy, color: "blue" } as IndexMapValue,
+	"Gaming": { svg: gaming, color: "gold" } as IndexMapValue,
+	"AI / Machine Learning": { svg: aiML, color: "blue" } as IndexMapValue,
+	"General Resources": { svg: tools, color: "gold" } as IndexMapValue,
+};
```
Comment (taesungh) [inline src/views/resources/Resources.tsx:40]: @RandyHuynh5815, this is a bit more complicated than I was expecting, but it will work for now.

**Review by taesungh — DISMISSED** [review]
Checks workflow didn't run since there are merge conflicts, but functionality alright :/

File: src/views/resources/Resource/Resource.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import { useRef } from "react";
```
Comment (njhuey) [inline src/views/resources/Resource/Resource.tsx:1]: Unused import

**Review by njhuey — DISMISSED** [review]

**Review by njhuey — APPROVED** [review]

**Review by taesungh — APPROVED** [review]

**Pull request merged**